src/epd5in65f/graphics.rs

```diff
@@ -0,0 +1,207 @@
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR
+                ))
+            );
+        }
+    }
+
+    #[test]
+    fn graphics_rotation_90() {
+        let mut display = Display5in65f::default();
+        display.set_rotation(DisplayRotation::Rotate90);
+
+        let _ = Line::new(
+            Point::new(0, WIDTH as i32 - 2),
+            Point::new(0, WIDTH as i32 - 1),
+        )
+        .into_styled(PrimitiveStyle::with_stroke(OctColor::Black, 1))
+        .draw(&mut display);
+
+        let buffer = display.buffer();
+
+        for &byte in buffer.iter().take(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((OctColor::Black, OctColor::Black))
+            );
+        }
+
+        for &byte in buffer.iter().skip(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR,
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR
+                ))
+            );
+        }
+    }
+
+    #[test]
+    fn graphics_rotation_180() {
+        let mut display = Display5in65f::default();
+        display.set_rotation(DisplayRotation::Rotate180);
+
+        let _ = Line::new(
+            Point::new(WIDTH as i32 - 2, HEIGHT as i32 - 1),
+            Point::new(WIDTH as i32 - 1, HEIGHT as i32 - 1),
+        )
+        .into_styled(PrimitiveStyle::with_stroke(OctColor::Black, 1))
+        .draw(&mut display);
+
+        let buffer = display.buffer();
+
+        for &byte in buffer.iter().take(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((OctColor::Black, OctColor::Black))
+            );
+        }
+
+        for &byte in buffer.iter().skip(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR,
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR
+                ))
+            );
+        }
+    }
+
+    #[test]
+    fn graphics_rotation_270() {
+        let mut display = Display5in65f::default();
+        display.set_rotation(DisplayRotation::Rotate270);
+
+        let _ = Line::new(
+            Point::new(HEIGHT as i32 - 1, 0),
+            Point::new(HEIGHT as i32 - 1, 1),
+        )
+        .into_styled(PrimitiveStyle::with_stroke(OctColor::Black, 1))
+        .draw(&mut display);
+
+        let buffer = display.buffer();
+
+        for &byte in buffer.iter().take(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((OctColor::Black, OctColor::Black))
+            );
+        }
+
+        for &byte in buffer.iter().skip(1) {
+            assert_eq!(
+                OctColor::split_byte(byte),
+                Ok((
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR,
+                    epd5in65f::DEFAULT_BACKGROUND_COLOR
+                ))
+            );
+        }
+    }
+}
```

Comment:
Could you please  add some simple check for the colors in the buffer here so we have tests whenever we want to change anything around the general octocolor parts?

Reply:
Added some tests.